src/lib/create.ts

```diff
@@ -1,40 +1,34 @@
-import {
-  CreateTableOptions,
-  CreateTableReceipt,
-  Connection,
-} from "../interfaces.js";
+import { Connection, CreateTableReceipt } from "../interfaces.js";
 import * as tablelandCalls from "./tableland-calls.js";
 import { registerTable } from "./eth-calls.js";
 import { BigNumber } from "ethers";
 /**
  * Registers an NFT with the Tableland Ethereum smart contract, then uses that to register
  * a new Table on Tableland
- * @param {string} query SQL create statement. Must include 'id' as primary key.
- * @param {CreateTableOptions} options List of options
- * @returns {string} The token ID of the table created
+ * @param {string} query SQL create statement.
+ * @returns {string} The smart contract transaction receipt.
  */
 export async function create(
   this: Connection,
-  query: string,
-  options: CreateTableOptions = {}
+  schema: string,
+  prefix: string = ""
 ): Promise<CreateTableReceipt> {
-  const authorized = await tablelandCalls.checkAuthorizedList.call(this);
-  if (!authorized) throw new Error("You are not authorized to create a table");
-  // Validation
+  // TODO: This is realted to issue#22, we might end up doing something like `await this.provider.getNetwork();`
+  const providerNetwork = await this.signer.provider?.getNetwork();
+  const chainId = providerNetwork?.chainId;
+  if (!chainId) throw new Error("cannot create table without provider network");
 
+  const query = `CREATE TABLE ${prefix}_${chainId} (${schema});`;
   // This "dryrun" is done to validate that the query statement is considered valid.
   // We check this before minting the token, so the caller won't succeed at minting a token
   // then fail to create the table on the Tableland network
-  await tablelandCalls.create.call(this, query, "1", {
-    dryrun: true,
-    ...options,
-  });
+  await tablelandCalls.hash.call(this, query);
 
-  let id = options.id;
-  if (!id) {
-    const { tableId } = await registerTable.call(this);
-    id = BigNumber.from(tableId).toString();
-  }
+  const txn = await registerTable.call(this, query);
 
-  return await tablelandCalls.create.call(this, query, id, options);
+  const [, event] = txn.events ?? [];
```

Comment:
Since we're getting the transaction anyway, why not just extract the stuff from the event? The validator might not have picked it up yet, but that's ok... we have the transaction txnHash to use to check for status later.